### DISCUSSION TITLE
Retention limited by available storage by oldest or by camera

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#5903] [Ask A Question]
Forgive me if it's in the docs but I couldn't find an answer if it is 

When storage is running low then Frigate will delete the oldest recording segments to make room for newer ones. Dows it do this on a camera by camera basis or all recordings lumped together and simply deletes the oldest events 

In other words if my front camera is set to retain events for 14 days and all my other cameras for 7 days and frigate starts to run low on storage will all deletes happen on the front camera as they will nominally always be the oldest?

**Comment by NickM-27** [comment]
It is based purely on the age of the recording segments, deleting oldest to newest until enough room has been recovered

**Reply by ediguidedog** [reply]
Ok thanks. I'm not sure that's the optimal way as generally the longest retained camera is usually the most important one. Would it not be better to base deletions on closeness to expiry (or even just the oldest xx mins for each camera)  and that way deletions will be shared evenly across all cameras rather than (as in my example) loosing upto 50% of my most important recordings before other cameras share the burden.

Of course the answer is more storage so I don't hit the limits but if we have to delete events would the above be more optimal?

**Reply by NickM-27** [reply]
I disagree. If the recordings are older then they're less crucial (more time to review or hold them) than another cameras. There's also the fact that different cameras have different bandwidths so you could end up with one camera being entirely wiped out on each run.

**Reply by NickM-27** [reply]
At the end of the day, this is just a safety feature and you should be configuring frigate so that it is not configured to hold more recordings than you have storage available

**Reply by ediguidedog** [reply]
Maybe we have different use cases. I have a much longer retention on my front camera as its one I may need to review days or weeks after the event for something as mundane as a missed parcel or caller. The other cameras would generally be there for home security so would be reviewed as soon as something untoward happened.  

I can see your point if we deleted the last x mins from all cameras and a specific camera only had one event that had literally just happened it would be wiped out so how about manipulating the retention times and letting the normal clean up routines do the work?  They could be temporarily dropped by one day each (subject to a minimum) and that way the load would be shared but individual cameras recent events would be protected.

**Reply by NickM-27** [reply]
We've spent a lot of time thinking about the logic, I would not necessarily count on it changing, especially not for 0.12 since it's already in RC. 

Like I said, you should really just configure frigate to retain an amount of days that actually fits on your storage. This function exists as a safeguard to ensure frigate can keep recording in the event that storage fills up unexpectedly, instead of crashing like it does in 0.11

**Reply by ediguidedog** [reply]
OK Understood. Thanks for taking the time to reply 

I guess I need to be a little cuter with the way I manage recordings (or get a bigger SSD). The main problem I have is if the weather is nice we spend much of our time in the garden so that generates a lot of events that are largely disposable. I realise we could use some sort of presence to stop the recordings happen but that would not help when we are at home and we do want the garden monitored.  In my workshop I can manage this if the radio is on by sending a MQTT event to stop the recordings but I can't see a simple way of achieving the same for the garden. If events were deleted as I described it would help a lot so maybe something for the future?
